clients/v1_repl/v1_repl.go

```diff
@@ -0,0 +1,858 @@
+	fmt.Printf("%s", color.GreenString("> "))
+	for scanner.Scan() {
+		c.executor(scanner.Text())
+		fmt.Printf("%s", color.GreenString("> "))
+	}
+	return nil
+}
+
+func (c *Client) gopher() {
+	color.Cyan(Gopher)
+}
+
+// The logic for the main prompt that is run in the REPL loop
+func (c *Client) executor(cmd string) {
+	if cmd == "" {
+		return
+	}
+	defer c.writeCommandToHistory(cmd)
+	cmdArgs := strings.Split(cmd, " ")
+	switch strings.ToLower(cmdArgs[0]) {
+	case "quit", "exit":
+		color.HiBlack("Goodbye!")
+		os.Exit(0)
+	case "gopher":
+		c.gopher()
+	case "node":
+		color.Yellow("The 'node' command is enterprise only, not available in the influx 2.x CLI - were you looking for the 1.x InfluxDB CLI?")
+	case "consistency":
+		color.Yellow("The 'consistency' command is not available in the influx 2.x CLI - were you looking for the 1.x InfluxDB CLI?")
+	case "help":
+		c.help()
+	case "history":
+		color.HiBlack(strings.Join(c.readHistory(), "\n"))
+	case "format":
+		c.setFormat(cmdArgs)
+	case "precision":
+		c.setPrecision(cmdArgs)
+	case "settings":
+		c.settings()
+	case "pretty":
+		c.togglePretty()
+	case "use":
+		c.use(cmdArgs)
+	case "insert":
+		c.insert(cmd)
+	case "clear":
+		c.clear(cmd)
+	default:
+		c.runAndShowQuery(cmd)
+	}
+}
+
+// Create a regex string for a named InfluxQL identifier, quoted or unquoted
+func identRegex(name string) string {
+	return `((?P<` + name + `>\w+)|(\"(?P<` + name + `_quote>.+)\"))`
+}
+
+// Get the value of a named InfluxQL identifier from a regex match map.
+// Returns empty string if no match
+func getIdentFromMatches(matches *map[string]string, name string) string {
+	if val, ok := (*matches)[name]; ok && val != "" {
+		return val
+	} else if val, ok := (*matches)[name+"_quote"]; ok && val != "" {
+		return val
+	}
+	return ""
+}
+
+// Create a regex match map from a regexp with named groups.
+// Returns nil if no match.
+func reSubMatchMap(r *regexp.Regexp, str string) *map[string]string {
+	match := r.FindStringSubmatch(str)
+	if match == nil {
+		return nil
+	}
+	subMatchMap := make(map[string]string)
+	for i, name := range r.SubexpNames() {
+		if i != 0 {
+			subMatchMap[name] = match[i]
+		}
+	}
+	return &subMatchMap
+}
+
+var insertIntoRegex string = `^(?i)INSERT INTO ` + identRegex("db") + `(\.` + identRegex("rp") + `)? (?P<point>.+)$`
+var insertRegex string = `^(?i)INSERT (?P<point>.+)$`
+
+func (c Client) insert(cmd string) {
+	var db string
+	var rp string
+	var point string
+	insertRgx := regexp.MustCompile(insertRegex)
+	insertIntoRgx := regexp.MustCompile(insertIntoRegex)
+	insertMatches := reSubMatchMap(insertRgx, cmd)
+	insertIntoMatches := reSubMatchMap(insertIntoRgx, cmd)
+	if insertIntoMatches != nil {
+		db = getIdentFromMatches(insertIntoMatches, "db")
+		rp = getIdentFromMatches(insertIntoMatches, "rp")
+		point = getIdentFromMatches(insertIntoMatches, "point")
+		if db != "" && rp == "" {
+			defaultRp, err := c.getDefaultRetentionPolicy(context.Background(), db)
```

Comment:
In 2.x we write the line protocol to a bucket, so in order to find the correct bucket to write to, I need to reverse search the default retention policy from the DBRPs because there is no actual "INSERT" command implemented at the 1.x query endpoint.

I don't know of a way to do that (db,rp)->bucket mapping without querying for the default rp if it's empty. Any ideas?

Reply:
Oh, yep, the reason I was doing it was because I was using the 2.x write endpoint and trying to generate a (db,rp)->bucket mapping when I didn't realize there was a 1.x legacy write endpoint right there. I'll take this out.